test_agent/rust/.gitignore

```diff
@@ -0,0 +1 @@
+/target
```

Comment:
Similar to my comment [here](https://github.com/eclipse-uprotocol/up-tck/pull/23/files#r1565791884), I would also make this its own crate. This time a binary crate.

From within this folder you can run:
```
cargo init --bin test-agent
```

to get a folder test-agent containing the typical Cargo.toml and src/main.rs.

As I was mentioning [in the other comment](https://github.com/eclipse-uprotocol/up-tck/pull/23/files#r1565791884) you can then add to this Cargo.toml this line:
```
[dependencies]
utransport_socket = { path = "../../up_client_socket/rust/utransport_socket" }
```
(again, I probably got the number of ".." wrong)

I would then start by copying in the dependencies from your outer Cargo.toml and delete that one as its current usage is not idiomatic.